src/project.cc

```diff
@@ -402,21 +417,26 @@ Project::Entry GetCompilationEntryFromCompileCommandEntry(
   for (const auto& flag : config->extra_flags)
     result.args.push_back(flag);
 
-  // Add -resource-dir so clang can correctly resolve system includes like
-  // <cstddef>
-  if (!AnyStartsWith(result.args, "-resource-dir"))
-    result.args.push_back("-resource-dir=" + config->resource_dir);
-
-  // There could be a clang version mismatch between what the project uses and
-  // what cquery uses. Make sure we do not emit warnings for mismatched options.
-  if (!AnyStartsWith(result.args, "-Wno-unknown-warning-option"))
-    result.args.push_back("-Wno-unknown-warning-option");
-
-  // Using -fparse-all-comments enables documentation in the indexer and in
-  // code completion.
-  if (g_config->index.comments > 1 &&
-      !AnyStartsWith(result.args, "-fparse-all-comments")) {
-    result.args.push_back("-fparse-all-comments");
+  if (compiler_type) {
```

Comment:
I think this section should be based on if we are in cl parsing mode, not what the compiler type is. iirc, we always pass these arguments to libclang and never to a different compiler.

Reply:
I am not sure what changes should be made.

Further, I think compiler_type should include the information about cl mode.
How about make the compiler type like:
```cpp
enum class CompilerType { Unknown = 0, Clang = 1, GCC = 2, MSVC = 3, ClangCl = 4 };
```
Then the cl mode is equivalent to `compiler_type == CompilerType::MSVC ||  compiler_type == CompilerType::ClangCl`.